src/components/Sidebar.tsx

```diff
@@ -63,15 +63,24 @@ export const Sidebar: React.FC = () => {
           <>
             <button
               className={footerButtonClasses}
-              onClick={fetchNotifications}
+              onClick={() => {
+                history.replace('/');
+                fetchNotifications;
+              }}
               aria-label="Refresh Notifications"
             >
               <IconRefresh className="w-3.5 h-3.5" />
             </button>
 
             <button
               className={footerButtonClasses}
-              onClick={() => history.push('/settings')}
+              onClick={() => {
+                if (window.location.hash.includes('settings')) {
```

Comment:
To check the current location you should use `react-router-dom`'s...

```ts
const location = useLocation();
```

And then:

```ts
if (location.pathname.startsWith('/settings')) {
```

(I think)

Reply:
@hughlilly did you try the above approach?